packages/rangelink-vscode-extension/scripts/orchestrate-release-lock.sh

```
@@ -0,0 +1,112 @@
+QA_ISSUE_URL=$(echo "$QA_OUTPUT" | grep -o 'https://github\.com/[^ ]*issues/[0-9]*')
+if [[ -z "$QA_ISSUE_URL" ]]; then
+  echo -e "${RED}Error: could not extract QA issue URL from generate-qa-issue.sh output${NC}" >&2
+  exit 1
+fi
+
+# --- Step 3: Generate commit message ---
+
+COMMIT_MSGS_DIR="$REPO_ROOT/.commit-msgs"
+mkdir -p "$COMMIT_MSGS_DIR"
+
+LAST=$(ls "$COMMIT_MSGS_DIR" | grep -o '^[0-9]\{4\}' | sort -n | tail -1)
```

Shellcheck (0.11.0)

[warning] Don't use ls | grep. Use a glob or a for loop with a condition to allow non-alphanumeric filenames.

(SC2010)